components/frontend/src/header_footer/CollapseButton.test.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { fireEvent, render, renderHook, screen } from '@testing-library/react';
+import history from 'history/browser';
+import { CollapseButton } from './CollapseButton';
+import { useExpandedItemsSearchQuery } from '../app_ui_settings';
+import { createTestableSettings } from '../__fixtures__/fixtures';
+
+beforeEach(() => {
+    history.push("")
+});
+
+function renderCollapseButton(
+    {
+        expandedItems = null
+    } = {}
+) {
+    const settings = createTestableSettings()
+    render(
+        <CollapseButton
+            expandedItems={expandedItems ?? settings.expandedItems}
+        />
+    );
+}
+
+it("resets the expanded items", () => {
+    history.push("?expanded=tab")
+    const expandedItems = renderHook(() => useExpandedItemsSearchQuery())
+    expect(expandedItems.result.current.value).toStrictEqual(["tab"])
```

Comment:
Shouldn't we also test that opening the page with expanded=item does correctly display the item expanded?

Reply:
Or is this covered in `widgets/HeaderWithDetails.test.js`?

Reply:
Not really, I've added a test.